dart/dynamics/MetaSkeleton.hpp

```diff
@@ -95,12 +95,41 @@ class MetaSkeleton : public common::Subject
   /// Get const BodyNode whose index is _idx
   virtual const BodyNode* getBodyNode(std::size_t _idx) const = 0;
 
+  /// Returns the body node of given name.
+  ///
+  /// When there are multiple body nodes with the same name, returns the first
+  /// body node, which is implementation dependent, and prints a warning.
```

Comment:
Nit: Add a note about how this can occur, i.e. when the `MetaSkeleton` contains `BodyNode`s from multiple `Skeleton`s. The same comment applies to the other three functions you added.